src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,349 @@
+                                       Data_Import_Field_Mapping__mdt existingDefaultFieldMapping) {
+
+        fieldMapping.targetObjectMappingLabel = existingDefaultObjectMapping.Label;
+        fieldMapping.targetObjectMappingRecordName = existingDefaultObjectMapping.DeveloperName;
+        fieldMapping.dataImportFieldMappingSetLabel = defaultFieldMappingSet.Label;
+        fieldMapping.dataImportFieldMappingSetRecordName = defaultFieldMappingSet.DeveloperName;
+
+        if (existingDefaultFieldMapping != null) {
+            fieldMapping.recordName = existingDefaultFieldMapping.DeveloperName;
+        }
+    }
+
+    /**
+     * @description Check if Field Mapping Metadata Type field values are out of sync with default values
+     * This method isn't currently used anywhere.
+     * @return Boolean
+     */
+    private Boolean foundDefaultFieldMappingMismatch(BDI_HelpTextCMT.DataImportFieldMapping newFieldMapping,
+                                                     Data_Import_Field_Mapping__mdt existingFieldMapping,
+                                                     Data_Import_Object_Mapping__mdt existingDefaultObjectMapping) {
+
+        Boolean badMappingSetId = (existingFieldMapping.Data_Import_Field_Mapping_Set__c != defaultFieldMappingSet.Id);
+        Boolean badRequiredField = (existingFieldMapping.Required__c != 'No');
+        Boolean badSourceFieldAPIName = (existingFieldMapping.Source_Field_API_Name__c != newFieldMapping.sourceFieldAPIName);
+        Boolean badTargetFieldAPIName = (existingFieldMapping.Target_Field_API_Name__c != newFieldMapping.targetFieldAPIName);
+        Boolean badTargetObjectMappingId = (existingFieldMapping.Target_Object_Mapping__c != existingDefaultObjectMapping.Id);
+        Boolean badIsDeletedField = (existingFieldMapping.Is_Deleted__c != false);
+        Boolean foundMismatch = badMappingSetId || badRequiredField || badSourceFieldAPIName || badTargetFieldAPIName || badTargetObjectMappingId || badIsDeletedField;
+
+        return foundMismatch;
+    }
+
+    /**
+     * @description Go through all fields in Data Import object and collect relevant help text mappings
+     * @return A collection of help text field names mapped to an instance of MappingDetail which holds relevant values for building
+     * the corresponding Metadata Type record (i.e. Source Object Name, Target Object Name, Target Field API Name, etc)
+     */
+    public Map<String, BDI_HelpTextCMT.MappingDetail> getAllMappedHelpTextFields() {
+        Map<String, BDI_HelpTextCMT.MappingDetail> helpTextFieldMap = new Map<String, BDI_HelpTextCMT.MappingDetail>();
+        Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts> importStatusAndLookupMap = new Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts>();
+        Map<String, Schema.DescribeFieldResult> allDFRs = UTIL_Describe.getAllFieldsDescribe('npsp__DataImport__c');
```

Comment:
Move 'npsp__DataImport__c' to a defined constant? Also looks like this might fail in non-namespaced orgs (although I pretty much universally fail to predict that accurately)